packages/react-dom/src/__tests__/refs-test.js

```diff
@@ -351,6 +351,36 @@ describe('ref swapping', () => {
       'Expected ref to be a function, a string, an object returned by React.createRef(), or null.',
     );
   });
+
+  // @gate !__DEV__ || warnAboutCallbackRefReturningFunction
```

Comment:
The test passes in prod even without the flag because the assertions are only about the warning. Hence the condition.